pkg/R/RDD.R

```diff
@@ -64,43 +80,62 @@ setMethod("initialize", "PipelinedRDD", function(.Object, prev, func, jrdd_val)
     .Object@prev_jrdd <- getJRDD(prev)
     # Since this is the first step in the pipeline, the prev_serialized
     # is same as serialized here.
-    .Object@env$prev_serialized <- .Object@env$serialized
+    .Object@env$prev_serializedMode <- prev@env$serializedMode
+
+    .Object@colNames <- prev@colNames
   } else {
     pipelinedFunc <- function(split, iterator) {
       func(split, prev@func(split, iterator))
     }
     .Object@func <- pipelinedFunc
     .Object@prev_jrdd <- prev@prev_jrdd # maintain the pipeline
-    # Get if the prev_jrdd was serialized from the parent RDD
-    .Object@env$prev_serialized <- prev@env$prev_serialized
+    # Get the serialization mode of the parent RDD
+    .Object@env$prev_serializedMode <- prev@env$prev_serializedMode
+    
+    .Object@colNames <- prev@colNames
   }
 
   .Object
 })
 
-
 #' @rdname RDD
 #' @export
 #'
 #' @param jrdd Java object reference to the backing JavaRDD
-#' @param serialized TRUE if the RDD stores data serialized in R
+#' @param serialized "byte" if the RDD stores data serialized in R, "string" if the RDD stores strings, 
+#'        and "row" if the RDD stores the rows of a DataFrame
 #' @param isCached TRUE if the RDD is cached
 #' @param isCheckpointed TRUE if the RDD has been checkpointed
-RDD <- function(jrdd, serialized = TRUE, isCached = FALSE,
-                isCheckpointed = FALSE) {
-  new("RDD", jrdd, serialized, isCached, isCheckpointed)
+RDD <- function(jrdd, serializedMode = "byte", isCached = FALSE,
+                isCheckpointed = FALSE, colNames = list()) {
+  new("RDD", jrdd, serializedMode, isCached, isCheckpointed, colNames)
 }
 
 PipelinedRDD <- function(prev, func) {
   new("PipelinedRDD", prev, func, NULL)
 }
 
+# Return the serialization mode for an RDD.
+setGeneric("getSerializedMode", function(rdd, ...) { standardGeneric("getSerializedMode") })
+# For normal RDDs we can directly read the serializedMode
+setMethod("getSerializedMode", signature(rdd = "RDD"), function(rdd) rdd@env$serializedMode )
+# For pipelined RDDs if jrdd_val is set then serializedMode should exist
+# if not we return the defaultSerialization mode of "byte" as we don't know the serialization
+# mode at this point of time.
+setMethod("getSerializedMode", signature(rdd = "PipelinedRDD"),
+          function(rdd, defaultSerialization = "byte") {
```

Comment:
The argument `defaultSerialization` doesn't seem intended to be supplied by a caller -- makes sense to just move it inside the function body?